grails-app/controllers/au/org/ala/profile/hub/ProfileController.groovy

```diff
@@ -381,6 +381,19 @@ class ProfileController extends BaseController {
         }
     }
 
+    def getPublication (){
+        render view: "publication";
+    }
+
+    def getPublicationJson(){
+        if(!params.pubId){
+            badRequest "Publication Id must be provided";
+        } else {
+            def result = profileService.getPublicationJson(params.pubId);
```

Comment:
If you return the same object that WebService.get returns, you can use the "handle" method on the base controller, which will deal with errors as well as the response.